dill/_dill.py

```diff
@@ -877,11 +922,15 @@ def __getattribute__(self, attr):
         return type(self)(attrs, index)
 
 if PY3:
-    def _create_cell(contents):
-        return (lambda y: contents).__closure__[0]
+    def _create_cell(contents=None):
+        return (lambda: contents).__closure__[0]
+
 else:
-    def _create_cell(contents):
-        return (lambda y: contents).func_closure[0]
+    def _create_cell(contents=None):
+        if contents is not _CELL_EMPTY:
```

Comment:
`_CELL_EMPTY` is defined by `dill._shims.Shim('_CELL_EMPTY')` adding `_CELL_EMPTY` to `dill._shims._dill_true_values`.  We may want to add a brief inline comment here to remind developers where `_CELL_EMPTY` comes from.  Similarly for any of the other objects added to `dill._dill` in `dill._shims`.  Also, I think the namespace manipulation performed in `dill._shims` should be better documented (inline).

Reply:
Doc LGTM